redis/redis.test.w

```diff
@@ -0,0 +1,21 @@
+bring expect;
+bring "./redis.w" as r;
+
+let redis = new r.Redis();
+
+test "set and get" {
+  redis.set("foo", "bar");
+  expect.equal(redis.get("foo"), "bar");
+}
+
+test "set, get and del" {
```

Comment:
The same thing happening here as well. But the IoRedis client is looking for the container with the port it's first initialized with which doesn't exist anymore, causing `[ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:32775`

Reply:
> But the IoRedis client is looking for the container with the port it's first initialized

why is it using the previous port?

Reply:
Hi, it's making a socket connection with the server, and when server is not there, it's keep looking for the server. I've figured out a way though, it's working but, 
```js
let port = url.split(":")[2];
  let redis = new IoRedis(port)
  redis.on("error", (err) => {  // by this, we can listen if there's any error
    console.log(`Redis error: ${err}`);
    redis.disconnect(); // this will disconnect the connection between server and the client, and it will stop looking for server. 
  });
  return redis;
```
But the problem is `redis.disconnect()` is disconnecting all the available clients. 

During wing test, wing is creating multiple redis container, therefore multiple clients has been created with diff port, and when one of the container is getting killed and hitting the "error" event to trigger `redis.disconnect()`, it is deleting all the client beforehand even they start executing the tests, so the error is coming `connection is closed` for next 2 tests (first tests passing successfully)

So, now I think the only way to somehow disconnect only one client at a time. But I'm not able to figure out how. 
somthing like this. 
```js 
redis.disconnect(port); // though it is not a valid 
```
Can you please let me know if you have any suggestions, idea on how can I make this work, though I've just joined official redis discord and put this query to their channel, hope they help.

Reply:
I am not sure I fully understand what's going on here. If for every test we start a new container on a new port and there's a client that connects to that server, why are they all disconnected together?

Does this problem also exists in our current [`ex` implementation](https://github.com/winglang/wing/blob/main/libs/wingsdk/src/target-sim/redis.inflight.ts)? 

Copying @yoav-steinberg who wrote the `ex` implementation.

Reply:
Hi, I have found the problem.

When I'm running the test at wing-console, it is passing successfully and at each test I'm running shell command `docker ps`, and you can see the log, the url port is same as the docker instance is running.

![Screenshot from 2023-11-26 13-57-47](https://github.com/winglang/winglibs/assets/75121304/827b96ac-5127-4d80-92ad-2bbaf2fc8d45)

But the same test file when I'm running by the command from terminal `wing test redis.test.w`, as you can see the url port is different than the docker instance running, that specific docker instance is closed in this case.
![image](https://github.com/winglang/winglibs/assets/75121304/f08f83f5-206f-46f7-80f7-b8ad80cb9594)
 
I think this is a problem with tests, let me know what can be a solution of that. I was wrong about that the problem is redis client specific but it is not.